core/src/surplus.ts

```diff
@@ -0,0 +1,52 @@
+import type { SurplusAuctionData, SurplusAuctionStates } from './types';
+import { getEarliestDate } from './helpers/getEarliestDate';
+import BigNumber from './bignumber';
+import getContract from './contracts';
+
+export const fetchSurplusAuctionByIndex = async function (
+    network: string,
+    auctionIndex: number
+): Promise<SurplusAuctionData> {
+    const contract = await getContract(network, 'MCD_FLAP');
+
+    const auctionsQuantityBinary = await contract.kicks();
+    const auctionsQuantity = new BigNumber(auctionsQuantityBinary._hex);
+    if (auctionsQuantity.lt(auctionIndex)) {
+        throw new Error('No active auction found with this id');
+    }
+
+    const auctionData = await contract.bids(auctionIndex);
+    const isAuctionDeleted = new BigNumber(auctionData.lot).eq(0);
+    const baseAuctionInfo: SurplusAuctionData = {
```

Comment:
- I would suggest to maybe have a different type for the collected auction? Otherwise, types doesn't really make sense as most of the parameters are optional. Would it make sense?
- I would check that `end` date is 0 instead of `lot` to be on the safe side